zeth_contracts/contracts/Groth16Mixer.sol

```diff
@@ -17,14 +17,16 @@ import "./Pairing.sol";
 contract Groth16Mixer is BaseMixer {
 
     // The structure of the verification key differs from the reference paper.
-    // It doesn't contain any element of GT, but only elements of G1 and G2 (the
-    // source groups).  This is due to the lack of precompiled contract to
-    // manipulate elements of the target group GT on Ethereum.
+    // It doesn't contain any element of GT, but only elements of G1 and G2
+    // (the source groups). This is due to the lack of precompiled contract to
+    // manipulate elements of the target group GT on Ethereum. Note that Beta
+    // and Delta are negated to avoid having to perform point negations in
+    // contract code.
     struct VerifyingKey {
-        Pairing.G1Point Alpha;      // slots 0x00, 0x01
-        Pairing.G2Point Beta;       // slots 0x02, 0x03, 0x04, 0x05
-        Pairing.G2Point Delta;      // slots 0x06, 0x07, 0x08, 0x09
-        Pairing.G1Point[] ABC;      // slot 0x0a
+        Pairing.G1Point Alpha;       // slots 0x00, 0x01
+        Pairing.G2Point Minus_Beta;  // slots 0x02, 0x03, 0x04, 0x05
+        Pairing.G2Point Minus_Delta; // slots 0x06, 0x07, 0x08, 0x09
```

Comment:
Related to: https://github.com/clearmatics/zeth/pull/293#discussion_r505464883 I guess. Please make sure to document that in https://github.com/clearmatics/zeth/pull/293#discussion_r505464883 you manipulate -Beta and -Delta, otherwise this is confusing

Reply:
Added comments in the zksnark tests

Reply:
Thanks